rtengine/imageio.cc

```diff
@@ -967,26 +964,24 @@ int ImageIO::loadJxl(const Glib::ustring &fname)
         }
     }  // end grand decode loop
 
-    unsigned int width = info.xsize;
-    unsigned int height = info.ysize;
+    std::size_t width = info.xsize;
+    std::size_t height = info.ysize;
 
     allocate(width, height);
 
-    int line_length = width * 3 * 4;
+    std::size_t line_length = width * 3 * 4;
 
-    for (size_t row = 0; row < height; ++row) {
-        setScanline (row, ((const unsigned char*)buffer) + (row * line_length), 32);
+    for (std::size_t row = 0; row < height; ++row) {
+        setScanline(row, ((const unsigned char*)buffer.data()) + (row * line_length), 32);
```

Comment:
I don't think this C-style cast is needed anymore. That's why I suggested `buffer` should be `std::vector<unsigned char>`...

Reply:
I will try removing the cast.  If no problems, will bundle with other changes.